### PR TITLE
Flavor feature

### PR DESCRIPTION
**Pull request by aljesusg**
https://www.pivotaltracker.com/story/show/148535475

### CODE REVIEW

File: app/models/manageiq/consumption/showback_event.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ class ManageIQ::Consumption::ShowbackEvent < ApplicationRecord
 
   include_concern 'CPU'
   include_concern 'MEM'
-
+  include_concern 'TEMPLATE'
```
Comment (chargio) [inline app/models/manageiq/consumption/showback_event.rb]: TEMPLATE should be changes to another name (perhaps VM_TEMPLATE). It is not clear defined as tempalte

Suggestions:
ENVELOPE
RESOURCES_REQUESTED
FLAVOR

File: app/models/manageiq/consumption/showback_event.rb [hunk 2]
```diff
@@ -45,7 +45,7 @@ def generate_data
       next unless resource.type.ends_with?(measure_type.category)
       self.data[measure_type.measure] = {}
       measure_type.dimensions.each do |dim|
-        self.data[measure_type.measure][dim] = 0
+        self.data[measure_type.measure][dim] = 0 unless measure_type.measure == "TEMPLATE"
```
Comment (chargio) [inline app/models/manageiq/consumption/showback_event.rb]: This creates the necessity to test in every function that the template is different
[ [ value ] ]?
Normally only one would be needed.

File: db/fixtures/showback_usage_types.yml [hunk 3]
```diff
@@ -4,6 +4,10 @@
   :measure: "CPU"
   :dimensions: ["average", "number", "max_number_of_cpu"]
 - :category: "Vm"
+  :description: "Template Showback of VM"
+  :measure: "TEMPLATE"
+  :dimensions: ["number","memory_reserve"]
```
Comment (chargio) [inline db/fixtures/showback_usage_types.yml:9]: number? ===> cpu_reserved?

File: spec/models/showback_event/mem_spec.rb [hunk 4]
```diff
@@ -33,9 +33,9 @@
     end
 
     it "Calculate MEM_total_mem" do
-      expect(event.MEM_total_mem(1024)).to eq(1024)
-      expect(event.MEM_total_mem(0)).to eq(1024)
-      expect(event.MEM_total_mem(2048)).to eq(2048)
+      expect(event.MEM_total_mem(4096)).to eq(4096)
```
Comment (chargio) [inline spec/models/showback_event/mem_spec.rb]: Some comments are needed. I don't understasnt the test

**Review by chargio — CHANGES_REQUESTED** [review]
Changes needed to make it easier to be understood. Name "template" is confliciting with a template for the feature

File: spec/models/showback_event/mem_spec.rb (hunk 4 above)
Comment (aljesusg) [reply spec/models/showback_event/mem_spec.rb]: Thanks @sergio-ocon this is wrong, this calculate the max_memory used at month and not the actual :( I need change it

File: app/models/manageiq/consumption/showback_column_units.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+:cpu_usage_rate_average: "Vm"
+:numvcpus: "vcpus"
+:cpu_total_cores: "cores"
+:total_mem: "Mb"
```
Comment (chargio) [inline app/models/manageiq/consumption/showback_column_units.yml:5]: ¿total mem unit?

File: db/fixtures/showback_usage_types.yml (hunk 3 above)
Comment (chargio) [reply db/fixtures/showback_usage_types.yml:9]: You can use any unit, if it is not identified we won't be able to translate it to K,M, etc

**Review by chargio — APPROVED** [review]
LGTM

File: app/models/manageiq/consumption/showback_column_units.yml (hunk 5 above)
Comment (aljesusg) [reply app/models/manageiq/consumption/showback_column_units.yml:5]: I don't understand your question

Comment (chargio) [reply app/models/manageiq/consumption/showback_column_units.yml:5]: My mistake

**Pull request closed without merging**